hack/verify-golint.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT_DIR=$(dirname $0)/..
+cd ${REPO_ROOT_DIR}
+
+if [[ -z "$(which golint)" ]]; then
+  echo 'Can not find golint, install with:'
+  echo 'go get -u github.com/golang/lint/golint'
+  exit 1
+fi
+
+array_contains () {
```

Comment:
I suggest you simplify this function by using `grep` (even if a temp file is needed) instead of a loop for in-memory comparison.